controllers/gce/backends/backends.go

```diff
@@ -210,24 +210,43 @@ func (b *Backends) create(namedPort *compute.NamedPort, hcLink string, sp Servic
 	return b.Get(namedPort.Port)
 }
 
-// Add will get or create a Backend for the given port.
+// Add will get or create Backends for the given ports.
 // Uses the given instance groups if non-nil, else creates instance groups.
-func (b *Backends) Add(p ServicePort, igs []*compute.InstanceGroup) error {
-	// We must track the port even if creating the backend failed, because
-	// we might've created a health-check for it.
-	be := &compute.BackendService{}
-	defer func() { b.snapshotter.Add(portKey(p.Port), be) }()
-
-	var err error
+func (b *Backends) Add(svcPorts []ServicePort, igs []*compute.InstanceGroup) error {
```

Comment:
What do you think about renaming this func to 'Sync'?

Reply:
Sync will imply that we will add missing ones and delete extras. But this one adds and not deletes. Thats the reason I prefer add over sync

Reply:
But we also update backend services.